### PR TITLE
Add FontCollection tests and fix compat bugs

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/font.c [hunk 1]
```diff
@@ -159,6 +161,13 @@ GdipPrivateAddFontFile (GpFontCollection *font_collection,  GDIPCONST WCHAR *fil
 	if (!file)
 		return OutOfMemory;
 
+	fileHandle = fopen ((char *)file, "r");
```
Comment (hughbe) [inline src/font.c]: `GdipPrivateAddFontFile` returns `FileNotFound` if the file doesn't exist with GDI+. Previously libgdiplus returned `Ok`

File: src/font.c [hunk 2]
```diff
@@ -290,13 +299,13 @@ GdipGetFontCollectionFamilyList (GpFontCollection *font_collection, INT num_soug
 	if (font_collection->config)
 		gdip_createPrivateFontSet (font_collection);
 
-	for (i = 0; i < font_collection->fontset->nfont; i++) {
+	for (i = 0; i < num_sought && i < font_collection->fontset->nfont; i++) {
```
Comment (hughbe) [inline src/font.c]: Here (and below), libgdiplus ignored the `num_sought` parameter. This now matches GDI+ behaviour, obviously.

File: src/font.c [hunk 3]
```diff
@@ -1152,14 +1160,12 @@ GdipPrivateAddMemoryFont(GpFontCollection *fontCollection, GDIPCONST void *memor
 	}
 	close(f);
 
-	/* FIXME - this doesn't seems to catch "bad" (e.g. invalid) font files */
-	result = FcConfigAppFontAddFile (fontCollection->config, fontfile);
```
Comment (hughbe) [inline src/font.c]: It appears that GDI+ just returns Ok for invalid files (added a test for that). Previously, libgdiplus would seemingly non-deterministically return `FileNotFound`.

In both libraries the count stays at the previous number if the font file is invalid.

File: src/font.c [hunk 4]
```diff
@@ -159,6 +161,13 @@ GdipPrivateAddFontFile (GpFontCollection *font_collection,  GDIPCONST WCHAR *fil
 	if (!file)
 		return OutOfMemory;
 
+	fileHandle = fopen ((char *)file, "r");
+	if(!fileHandle) {
```
Comment (akoeplinger) [inline src/font.c]: nit: space before `(`

File: src/font.c [hunk 5]
```diff
@@ -159,6 +161,13 @@ GdipPrivateAddFontFile (GpFontCollection *font_collection,  GDIPCONST WCHAR *fil
...
+		GdipFree (file);
+		return FileNotFound;
+	}
+
+	fclose(fileHandle);
```
Comment (akoeplinger) [inline src/font.c]: nit: space before `(`

**Comment by hughbe** [comment]
I've addressed the PR feedback.

> Where did you get the test.ttf and test.otf fonts from, are they open source?

I got them from the [corefx-testdata](https://github.com/dotnet/corefx-testdata) repository. I recall that I created the font files via a tool on my PC.

**Comment by hughbe** [comment]
Yes I am!

**Pull request merged**